src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+                            # Mark the module "sufficient"
+                            columns[1] = 'sufficient'
+                            auth.write(columns.join('    ') + "\n")
+                        else
+                            # Write down the module as-is
+                            auth.write(line)
+                        end
+                    else
+                        # Too few columns, just write it down.
+                        auth.write(line)
+                    end
+                end
+            }
+            # Eventually deny all access if no sufficient module is satisfied
+            # Bad luck for smart card users, this Yast module will not support that scenario.
+            auth.write("auth    required    pam_deny.so\n")
+            auth.close
+
+            # For accounts, make it sufficient to match local account.
+            lines = IO.readlines('/etc/pam.d/common-account')
+            account = File.open('/etc/pam.d/common-account', 'w+', 0600)
+            lines.each { |line|
+                columns = line.split(/\s+/)
+                if !/\s*#/.match(line) && columns.length >= 3
+                    if columns[2] == 'pam_unix.so'
+                        account.write(columns.join('    ') + "\n")
+                        account.write("account    sufficient    pam_localuser.so\n")
+                    elsif columns[2] != 'pam_localuser.so'
+                       # exclude existing line about pam_localuser
+                       account.write(line)
+                    end
+                else
+                    account.write(line)
+                end
+            }
+            account.close
+        end
+
+        # Clear all configuration objects.
+        def clear
+            # Kerberos configuration
+            @krb_conf = {'include' => [], 'libdefaults' => {}, 'realms' => {}, 'domain_realms' => {}, 'logging' => {}}
+            @krb_pam = false
+            # LDAP configuration (/etc/ldap.conf)
+            @ldap_conf = {}
+            @ldap_pam = false
+            @ldap_nss = []
+            # SSSD configuration (/etc/sssd/sssd.conf)
+            @sssd_conf = {}
+            @sssd_pam = false
+            @sssd_nss = []
+            @sssd_enabled = false
+            # Make the basic SSSD configuration structure
+            sssd_lint_conf
+            # Auxialiry daemons and mkhomedir
+            # If you call aux_apply, autofs will be enabled if any of the following condition is met:
+            # - autofs_enabled == true
+            # - SSSD is providing automount
+            # - LDAP is providing automount
+            @autofs_enabled = false
+            @nscd_enabled = false
+            @mkhomedir_pam = false
+            # AD enrollment details
+            @ad_domain = ''
+            @ad_user = ''
+            @ad_ou = ''
+            @ad_pass = ''
+            @ad_overwrite_smb_conf = false
+        end
+
+        def initialize
```

Comment:
It's because such method is the most important in the class and defines how it behaves, what are the defaults, etc. see http://www.rubyist.net/~slagell/ruby/objinitialization.html